src/distances/pairwise.jl

```diff
@@ -29,3 +29,31 @@ function pairwise!(
 )
     return Distances.pairwise!(out, d, reshape(x, :, 1), reshape(y, :, 1); dims=1)
 end
+
+
+# Also defines the colwise method for abstractvectors
+
+
+function colwise(d::PreMetric, x::ColVecs)
+    Distances.colwise(d, x.X, x.X)
+end
+
+function colwise(d::PreMetric, x::RowVecs)
+    Distances.colwise(d, x.X', x.X')
+end
+
+function colwise(d::PreMetric, x::AbstractVector)
+    d.(x, x)
```

Comment:
I think
```suggestion
    return map(d, x, x)
```
would be simpler here - there's no need for the broadcasting machinery since the inputs have the same dimension.